llvm/lib/IR/RuntimeLibcalls.cpp

```diff
@@ -255,4 +255,9 @@ void RuntimeLibcallsInfo::initLibcalls(const Triple &TT) {
     }
     setLibcallName(RTLIB::MULO_I128, nullptr);
   }
+
+  if (TT.isSystemZ()) {
+    setLibcallName(RTLIB::FPROUND_F32_F16, "__truncsfhf2");
+    setLibcallName(RTLIB::FPEXT_F16_F32, "__extendhfsf2");
+  }
```

Comment:
Why do these names need to be set, aren't these the default?

Reply:
you may be right - as I wrote above I have not really tried this before. Would you happen to know how to build and link these?

Reply:
Hm, I see they default to the `__gnu_` functions in this file. Some targets (wasm, hexagon) manually set it to `__extendhfsf2` and `__truncsfhf2` in `*SelfLowering.cpp` but why do targets like x86 correctly lower to these as well without an override either in this file or in selflowering?

Reply:
Regarding how to build and link, they are in compiler-rt if that can be built https://github.com/llvm/llvm-project/blob/fa22100d57631bbb0a507dd27e3ebb24b1354623/compiler-rt/lib/builtins/truncsfhf2.c#L15. `__trunc` and `__extend` are what you want to emit here, I'm just not sure what exactly this file needs to do because it seems like `HasLegalHalfType` controls `__extend`/`__trunc` vs. `__gnu_` lowering somehow https://github.com/llvm/llvm-project/pull/109164#issuecomment-2433525551.